src/master-playlist-controller.js

```diff
@@ -729,15 +734,17 @@ export class MasterPlaylistController extends videojs.EventTarget {
   }
 
   shouldSwitchToMedia_(nextPlaylist) {
-    const currentPlaylist = this.masterPlaylistLoader_.media();
+    const currentPlaylist = this.masterPlaylistLoader_.media() ||
+      this.masterPlaylistLoader_.pendingMedia_;
     const buffered = this.tech_.buffered();
-    const forwardBuffer = buffered.length ?
-      buffered.end(buffered.length - 1) - this.tech_.currentTime() : 0;
-
+    const currentTime = this.tech_.currentTime();
+    const forwardBuffer = Ranges.timeAheadOf(buffered, currentTime);
     const bufferLowWaterLine = this.bufferLowWaterLine();
     const bufferHighWaterLine = this.bufferHighWaterLine();
+    const isBuffered = Boolean(Ranges.findRange(buffered, currentTime).length);
 
     return shouldSwitchToMedia({
+      isBuffered,
```

Comment:
We don't really want to switch at the start of playback for live streams before we have actually started. It almost doubles are time to first playback.